test/skate_web/controllers/detours_controller_test.exs

```diff
@@ -58,6 +58,59 @@ defmodule SkateWeb.DetoursControllerTest do
     end
   end
 
+  describe "get_detours/2" do
+    @tag :authenticated
+    test "fetches detours from database and groups by active, past, draft", %{conn: conn} do
+      conn =
+        put(conn, "/api/detours/update_snapshot", %{
+          "snapshot" => %{
+            "context" => %{
+              "route" => %{
+                "name" => "23",
+                "directionNames" => %{
+                  "0" => "Outbound",
+                  "1" => "Inbound"
+                }
+              },
+              "routePattern" => %{
+                "headsign" => "Headsign",
+                "directionId" => 0
+              },
+              "nearestIntersection" => "Street A & Avenue B",
+              "uuid" => 1
+            }
+          }
+        })
+
+      %Skate.Detours.Db.Detour{author_id: author_id_on_put} = Detours.get_detour!(1)
+
+      assert %{"data" => 1} = json_response(conn, 200)
+
+      conn = get(conn, "/api/detours/get_detours")
+
+      assert %{
+               "data" => %{
+                 "active" => nil,
+                 "draft" => [
+                   %{
+                     "direction" => "Outbound",
+                     "intersection" => "Street A & Avenue B",
+                     "name" => "Headsign",
+                     "route" => "23",
+                     "status" => "draft",
+                     "author_id" => author_id_on_get,
```

Comment:
```suggestion
                     "author_id" => ^author_id_on_put,
```

☝️ non-blocking, but then you don't need to have two author_id variables, and could probably even just rename it to `author_id`.